src/Microsoft.Health.Fhir.Core/Features/Definition/ISearchParameterDefinitionManager.cs

```diff
@@ -70,17 +71,17 @@ public interface ISearchParameterDefinitionManager
         /// <returns>The search parameter type.</returns>
         SearchParamType GetSearchParameterType(SearchParameterInfo searchParameter, int? componentIndex);
 
-        /// <summary>
-        /// Updates the existing resource type - search parameter hash mapping with the given new values.
-        /// </summary>
-        /// <param name="updatedSearchParamHashMap">Dictionary containing resource type to search parameter hash values</param>
-        public void UpdateSearchParameterHashMap(Dictionary<string, string> updatedSearchParamHashMap);
-
         /// <summary>
         /// Gets the hash of the current search parameters that are supported for the given resource type.
         /// </summary>
         /// <param name="resourceType">Resource type for which we need the hash of search parameters.</param>
         /// <returns>A string representing a hash of the search parameters.</returns>
         public string GetSearchParameterHashForResourceType(string resourceType);
+
+        /// <summary>
+        /// Allows addition of a new search parameters at runtime.
+        /// </summary>
+        /// <param name="searchParameters">An ennumeration containing SearchParameter resources.</param>
+        void AddNewSearchParameters(IList<ITypedElement> searchParameters);
```

Comment:
>IList [](start = 36, length = 5)

nit: `IList` has `Insert` and `RemoveAt` methods which allow to change list content. I would change it to IEnumerable, since maybe point of this collection is to pass something which can be enumerated but not specifically modifed.